src/vocgui/views/training_set_by_id_viewset.py

```diff
@@ -0,0 +1,41 @@
+from rest_framework import viewsets
+from django.db.models import Count, Q
+from vocgui.models import TrainingSet
+from vocgui.serializers import TrainingSetSerializer
+from vocgui.models import Discipline
+from .utils import check_group_object_permissions
+
+
+class TrainingSetByIdViewSet(viewsets.ModelViewSet):
```

Comment:
```suggestion
class TrainingSetByIdViewSet(viewsets.ReadOnlyModelViewSet):
```
I'd use a read only model view so we can obmit the `http_method_names` restriction.